string action
int32 tool_id
float32 theta
int32 request_id
float64 x
float64 y
float64 z
